cypress/integration/alerts_dashboard_flyout_spec.js

```diff
@@ -0,0 +1,287 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import React from 'react';
+import { INDEX, PLUGIN_NAME } from '../support/constants';
+import sampleAlertsFlyoutBucketMonitor from '../fixtures/sample_alerts_flyout_bucket_level_monitor.json';
+import sampleAlertsFlyoutQueryMonitor from '../fixtures/sample_alerts_flyout_query_level_monitor.json';
+
+const BUCKET_MONITOR = 'sample_alerts_flyout_bucket_level_monitor';
+const BUCKET_TRIGGER = 'sample_alerts_flyout_bucket_level_trigger';
+const QUERY_MONITOR = 'sample_alerts_flyout_query_level_monitor';
+const QUERY_TRIGGER = 'sample_alerts_flyout_query_level_trigger';
+
+describe('Alerts by trigger flyout', () => {
+  before(() => {
+    // Delete any existing monitors
+    cy.deleteAllMonitors();
+
+    // Load sample data
+    cy.loadSampleEcommerceData();
+
+    // Create the test monitors
+    cy.createMonitor(sampleAlertsFlyoutBucketMonitor);
+    cy.createMonitor(sampleAlertsFlyoutQueryMonitor);
+    cy.wait(10000);
+
+    // Visit Alerting OpenSearch Dashboards
+    cy.visit(`${Cypress.env('opensearch_dashboards')}/app/${PLUGIN_NAME}#/monitors`);
+
+    // Confirm test monitors were created successfully
+    cy.contains(BUCKET_MONITOR);
+    cy.contains(QUERY_MONITOR);
+
+    // Wait 1.5 minute for the test monitors to trigger alerts, then go to the 'Alerts by trigger' dashboard page to view alerts
+    cy.wait(90000);
```

Comment:
Have you ever seen flakiness on this part before? I am just thinking if we are able to query for "x alerts" text on the alerts by triggers page instead. I believe on github actions it sometimes takes a bit longer for the individual steps to complete comparing to running test locally.

Reply:
I haven't seen any flakiness locally or on the PR checks so far. Just to confirm, are you referring to when the flyout link is clicked? I was reluctant to search for "x alerts" and click on that as I was thinking there could be times when there would be multiple instances of "1 alert" appearing on the dashboard.

Reply:
As per offline conversation, I'll be refactoring this `wait()` logic to instead search for the test monitors' names to ensure the dashboard page has loaded.